src/main/java/net/imagej/patcher/HeadlessGenericDialog.java

```diff
@@ -100,6 +100,21 @@ public HeadlessGenericDialog() {
 		radioButtons = new ArrayList<String>();
 	}
 
+	public <E extends Enum<E>> void addEnumChoice(String label, Enum<E> defaultItem) {
+		Class<E> enumClass = defaultItem.getDeclaringClass();
+		E[] enums = enumClass.getEnumConstants();
+		String[] items = new String[enums.length];
+		for (int i = 0; i < enums.length; i++) {
+			items[i] = enums[i].name();
+		}
+		this.addChoice(label, items, defaultItem.name());
+	}
+
+	public <E extends Enum<E>> E getNextEnumChoice(Class<E> enumClass) {
+		String choiceString = this.getNextChoice();
+		return Enum.valueOf(enumClass, choiceString);
+	}
+
```

Comment:
Based upon the description below, I'm not entirely sure that duplicating this code from ImageJ1 is necessary. It appears a form of reverse inheritance is used, so the resulting `GenericDialog` now with a `HeadlessGenericDialog` super class, should still have an implementation of this method.
https://github.com/imagej/ij1-patcher/blob/f2e6597176d07863a25e1a6565709a1e381f0815/src/main/java/net/imagej/patcher/HeadlessGenericDialog.java#L51-L55